common/src/main/java/com/abrenoch/hyperiongrabber/common/util/BorderProcessor.java

```diff
@@ -0,0 +1,177 @@
+package com.abrenoch.hyperiongrabber.common.util;
+
+import java.nio.ByteBuffer;
+
+public class BorderProcessor {
+
+    private final int MAX_INCONSISTENT_FRAME_COUNT = 10;
+    private final int MAX_UNKNOWN_FRAME_COUNT = 600;
+    private final int BORDER_CHANGE_FRAME_COUNT = 50;
+
+    private final int BLACK_THRESHOLD; // unit to detect how black a pixel is [0..255]
+    private BorderObject mPreviousBorder;
+    private BorderObject mCurrentBorder;
+    private int mConsistentFrames = 0;
+    private int mInconsistentFrames = 0;
+
+    public BorderProcessor(int blackThreshold) {
+        BLACK_THRESHOLD = blackThreshold;
+    }
+
+    public BorderObject getCurrentBorder() { return mCurrentBorder; }
+
+    private void checkNewBorder(BorderObject newBorder) {
+        if (mPreviousBorder != null && mPreviousBorder.equals(newBorder)) {
+            ++mConsistentFrames;
+            mInconsistentFrames = 0;
+        } else {
+            ++mInconsistentFrames;
+
+            if (mInconsistentFrames <= MAX_INCONSISTENT_FRAME_COUNT) {
+                return;
+            }
+
+            mPreviousBorder = newBorder;
+            mConsistentFrames = 0;
+        }
+
+        if (mCurrentBorder != null && mCurrentBorder.equals(newBorder)) {
+            mInconsistentFrames = 0;
+            return;
+        }
+
+        if (!newBorder.isKnown()) {
+            if (mConsistentFrames == MAX_UNKNOWN_FRAME_COUNT) {
+                mCurrentBorder = newBorder;
+            }
+        } else {
+            if (mCurrentBorder == null || !mCurrentBorder.isKnown() ||
+                    mConsistentFrames == BORDER_CHANGE_FRAME_COUNT) {
+                        mCurrentBorder = newBorder;
+            }
+        }
+    }
+
+    public void parseBorder(ByteBuffer buffer, int width, int height, int rowStride,
+                            int pixelStride) {
+        checkNewBorder( findBorder(buffer, width, height, rowStride, pixelStride) );
+    }
+
+    private BorderObject findBorder(ByteBuffer buffer, int width, int height, int rowStride,
+                                    int pixelStride) {
+
+        int width33percent = width / 3;
+        int width66percent = width33percent * 2;
+        int height33percent = height / 3;
+        int height66percent = height33percent * 2;
+        int yCenter = height / 2;
+        int xCenter = width / 2;
+        int firstNonBlackYPixelIndex = -1;
+        int firstNonBlackXPixelIndex = -1;
+
+        int c1R, c1G, c1B;
+        int c2R, c2G, c2B;
+        int c3R, c3G, c3B;
+        int pos33p, pos66p;
+
+        buffer.position(0).mark();
+
+        for (int x = 0, posYCLp; x < width33percent; x++) {
```

Comment:
Never knew you could define multiple variiables in a for-loop, wow. Anyways, Id say it should be defined outside the loop accomapnied by a comment because I don't know what CLp means. In general I advise against abbreviations